addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/SerialOceanicThingHandler.java

```diff
@@ -0,0 +1,315 @@
+                        e.getMessage(), e);
+            }
+        }
+
+        IOUtils.closeQuietly(inputStream);
+        IOUtils.closeQuietly(outputStream);
+        if (serialPort != null) {
+            serialPort.close();
+        }
+
+        super.dispose();
+    }
+
+    @Override
+    protected String requestResponse(String commandAsString) {
+        synchronized (this) {
+            SerialOceanicBindingConfiguration config = getConfigAs(SerialOceanicBindingConfiguration.class);
+
+            Throttler.lock(config.port);
+
+            lastLineReceived = "";
+            String request = commandAsString + "\r";
+            String response = null;
+
+            try {
+                if (logger.isTraceEnabled()) {
+                    logger.trace("Requesting : {} ('{}')", request, request.getBytes());
+                }
+                outputStream.write(request.getBytes());
+                outputStream.flush();
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Error writing '" + request + "' to serial port " + config.port + " : " + e.getMessage());
+            }
+
+            long timeStamp = System.currentTimeMillis();
+            while (lastLineReceived.equals("")) {
+                try {
+                    Thread.sleep(50);
+                } catch (InterruptedException e) {
+                    logger.error("An exception occurred while putting the thread to sleep: {}", e.getMessage());
+                }
+                if (System.currentTimeMillis() - timeStamp > REQUEST_TIMEOUT) {
+                    logger.warn("A timeout occurred while requesting data from the water softener");
+                    readerThread.reset();
+                    break;
+                }
+            }
+            response = lastLineReceived;
+
+            Throttler.unlock(config.port);
+
+            return response;
+        }
+    }
+
+    public class SerialPortReader extends Thread {
+
+        private boolean interrupted = false;
+        private InputStream inputStream;
+        int index = 0;
+
+        SimpleDateFormat sdf = new SimpleDateFormat("HH:mm:ss,SSS");
+
+        public SerialPortReader(InputStream in) {
+            this.inputStream = in;
+            this.setName("SerialPortReader-" + getThing().getUID());
+        }
+
+        public void reset() {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Resetting the SerialPortReader");
+            }
+            index = 0;
+        }
+
+        @Override
+        public void interrupt() {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Interrupting the SerialPortReader");
+            }
+            interrupted = true;
+            super.interrupt();
+            try {
+                inputStream.close();
+            } catch (IOException e) {
```

Comment:
You already use the util method to silently close, maybe use it here as well

Reply:
Not sure I understood this one

Reply:
You already use `closeQuietly` should you use it here as well?